src/styles/createMixins.d.ts

```diff
@@ -2,13 +2,19 @@ import { Breakpoints } from './createBreakpoints';
 import { Spacing } from './spacing';
 import { StyleRules } from '../styles';
 
-export interface Mixins {
+interface CustomMixins {
+  // Use interface declaration merging to add custom mixins
+}
+
+export interface Mixins extends CustomMixins {
   gutters: (styles: React.CSSProperties) => React.CSSProperties;
   toolbar: React.CSSProperties;
 }
 
-export default function createMixins<T = {}>(
+export type MixinsOptions = Partial<Mixins> & CustomMixins;
+
```

Comment:
I'm not sure the intent of `CustomMixins`, but I think it better to just have `Mixins` since they will ultimately be merged into the `mixins` property.  Further, if we just use `Mixins`, then any custom mixins will be optional.  We could declare them separately like `Theme` and `ThemeOptions` if you want the user to have control over customization and required/optional.

Reply:
The reason for `CustomMixins` was specifically because that part of `MixinsOptions` should _not_ be partial, as I understand it. Whatever you pass as the `mixins` argument to `createMixins` will be spread on top of `gutters` and `toolbar`:

```js
export default function createMixins(breakpoints, spacing, mixins) {
  return {
    gutters: styles => { /*...*/ },
    toolbar: { /*...*/ },
    ...mixins,
  };
}
```

so in order to make sure all required properties of your custom mixin are present, they are _required_ here, because they won't have defaults filled in from any other source. Only `gutters` and `toolbars` can be optionally overridden.

For example, if you had

```ts
interface CustomMixins {
  foo: () => void
}
```
 
then you want this to be a type error:

```ts
const theme = createMuiTheme({
  mixins: {}, // no foo provided
})
```

otherwise this would pass type checking but lead to a runtime exception:

```ts
theme.mixins.foo() // foo is undefined!
```

Reply:
So making it more analogous to the Theme and ThemeOptions would also solve this problem and be a bit more obvious wouldn't it?

Reply:
I actually think `Theme` / `ThemeOptions` should work more like this 🙂  Right now it's a bit tricky to make a custom `Theme` because you have to make 2 separate augmentations:

```ts
interface ThemeOptions {
  x?: {}
  y?: {}
  z?: {}
}
interface Theme {
  x: {}
  y: {}
  z: {}
}
```

Wouldn't it be better if you could just say

```ts
interface CustomTheme {
  x: {}
  y: {}
  z: {}
}
```

and be done with it?

Reply:
Yes that is fine, I just want a consistent approach

Reply:
Actually after looking at your example for custom themes in the docs, I think you're right. It's a better pattern to augment `Theme` and  `ThemeOptions` separately, due to the fact that you may want to make a custom creation function which defaults some of the options. And likewise for `Mixins`.